community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/pipes/LoadCSVPipe.scala

```diff
@@ -103,7 +103,7 @@ case class LoadCSVPipe(source: Pipe,
       implicit val s = state
       val url = getImportURL(urlExpression(context).asInstanceOf[String], state.query)
 
-      val iterator: Iterator[Array[String]] = state.resources.getCsvIterator(url, fieldTerminator)
+      val iterator: Iterator[Array[String]] = state.resources.getCsvIterator(url, fieldTerminator, format match {case HasHeaders => true; case _ => false})
```

Comment:
The `LoadCsvPeriodicCommitObserver` needs to know if it should discount a header row or not.